src/platform/datapath_raw.c

```diff
@@ -160,6 +160,8 @@ CxPlatDataPathInitialize(
 
     Status = CxPlatDataPathRouteWorkerInitialize(*NewDataPath);
     if (QUIC_FAILED(Status)) {
+        CxPlatDpRawUninitialize(*NewDataPath);
```

Comment:
Can we do this cleanup (including the CxPlatSockPoolUninitialize call that is made when CxPlatDpRawInitialize returns error) in the Error path? It seems like this function started out with that sort of design, and then we deviated starting with the CxPlatDpRawInitialize call.

if (*NewDataPath != NULL) {
    if (DpRawInitialized) {
        CxPlatDpRawUninitialize(*NewDataPath);
    }
    if ((*NewDataPath)->SocketPool != NULL) {
        CxPlatSockPoolUninitialize(&(*NewDataPath)->SocketPool);
    }
    CxPlatRundownUninitialize(&(*NewDataPath)->SocketsRundown);
    CXPLAT_FREE(*NewDataPath, QUIC_POOL_DATAPATH);
    *NewDataPath = NULL;
}